index.js

```diff
@@ -2,74 +2,108 @@
 var isRegexp = require('is-regexp');
 var isPlainObj = require('is-plain-obj');
 
-module.exports = function (val, opts, pad) {
-	var seen = [];
-
-	return (function stringify(val, opts, pad) {
-		opts = opts || {};
-		opts.indent = opts.indent || '\t';
-		pad = pad || '';
-
-		if (val === null ||
-			val === undefined ||
-			typeof val === 'number' ||
-			typeof val === 'boolean' ||
-			typeof val === 'function' ||
-			isRegexp(val)) {
-			return String(val);
-		}
+var stringify = module.exports = function (val, opts, pad, seen) {
+	opts = opts || {};
```

Comment:
Need to use https://github.com/sindresorhus/object-assign as you modify the user-supplied object below.